media/js/base/core-datalayer.js

```diff
@@ -39,11 +40,13 @@ if (typeof Mozilla == 'undefined') {
                     var element = arguments[i]['gtm.element'];
                     var href = element.href;
 
-                    if (element.hostname === document.location.hostname) {
-                        var path = href.split(element.host)[1];
+                    if (element.hostname === hostname) {
+                        // remove host and locale from internal links
+                        var path = href.replace(/^(?:https?\:\/\/)(?:[^\/])*/, '');
                         var locale = path.split('/')[1];
```

Comment:
Unsure of how problematic it may be on the site (if at all), but this code assumes a locale is present if the hostnames match. (If somehow a locale were *not* present, the first part of the URL would be cut off, e.g. `https://www.mozilla.org/firefox/new/` would be reported as just `/new/`.

Since our locales follow a predictable format, would it be worth it to be explicit in the code (with another regex) when looking for a locale? Something like:

`var locale = path.match(/^\/(\w{2}\-\w{2}|\w{2})\//);`